lib/pilot/build_manager.rb

```diff
@@ -7,9 +7,11 @@ def upload(options)
 
       Helper.log.info "Ready to upload new build to TestFlight (App: #{app.apple_id})...".green
 
-      package_path = PackageBuilder.new.generate(apple_id: app.apple_id,
-                                                 ipa_path: config[:ipa],
-                                             package_path: "/tmp")
+      plist = FastlaneCore::IpaFileAnalyser.fetch_info_plist_file(config[:ipa])
+      package_path = FastlaneCore::IpaUploadPackageBuilder.new.generate(app_id: app.apple_id,
+                                                                      ipa_path: config[:ipa],
+                                                                  package_path: "/tmp",
+                                                                      platform: plist["DTPlatformName"])
```

Comment:
Would it make sense to fallback to the iOS as the default platform here? As far as I know, `plist` might be `nil` if the plist file can't be found, or if there are multiple plist files and the wrong one is taken. This should not crash if `plist` is nil here